lib/src/component_declaration/component_base.dart

```diff
@@ -444,6 +445,38 @@ abstract class UiStatefulComponent<TProps extends UiProps, TState extends UiStat
   // ----------------------------------------------------------------------
 }
 
+class _WarnOnModify<K, V> extends MapView<K, V> {
+
+  //Used to customize warning based on whether the data is props or state
+  bool isProps;
+
+  String message;
+
+  _WarnOnModify(Map componentData, bool isProps): super(componentData){
+    this.isProps = isProps;
+  }
+
+  @override
+  operator []=(K key, V value) {
+    if (isProps) {
+      message =
+        '''
+          props["$key"] was updated incorrectly. Never mutate this.props directly, as it can cause unexpected behavior; props must be updated only by passing in new values when rerendering this component.
```

Comment:
Our team follows the rules that lines should never exceed 120 characters in length, including comments and strings. You can set up a column marker in your IDE. These lines should all be wrapped manually onto new lines before they hit 120 characters.
```suggestion
          props["$key"] was updated incorrectly. Never mutate this.props directly, as it can cause unexpected behavior; 
          props must be updated only by passing in new values when re-rendering this component.
```